src/axom/primal/operators/winding_number.hpp

```diff
@@ -254,6 +256,196 @@ double winding_number(const Point<T, 2>& q,
+                                         nurbs_cache.getEndPoint(),
+                                         isOnCurve,
+                                         edge_tol);
+  }
+
+  double gwn = 0.0;
+
+  bool isOnThisCurve = false;
+  isOnCurve = false;
+
+  for(int n = 0; n < nurbs_cache.getNumKnotSpans(); ++n)
+  {
+    gwn +=
+      detail::bezier_winding_number_memoized(q, nurbs_cache, n, 0, 0, isOnThisCurve, edge_tol, EPS);
+    isOnCurve = isOnCurve || isOnThisCurve;
+  }
+
+  return gwn;
+}
+
+//! \brief Overload without optional return parameter
+template <typename T>
+double winding_number(const Point<T, 2>& q,
+                      const detail::NURBSCurveGWNCache<T>& nurbs_cache,
+                      double edge_tol = 1e-8,
+                      double EPS = 1e-8)
+{
+  bool dummy_isOnCurve;
+  return winding_number(q, nurbs_cache, dummy_isOnCurve, edge_tol, EPS);
+}
+
+/*!
+ * \brief Computes the GWN for a 2D point wrt an array of cached data for 2D NURBS curves
+ *
+ * \param [in] q The query point to test
+ * \param [in] c_arr The array of curve objects
+ * \param [in] isOnCurve Set to true is the query point is on the curve
+ * \param [in] edge_tol The physical distance level at which objects are considered indistinguishable
+ * \param [in] EPS Miscellaneous numerical tolerance level for nonphysical distances
+ * 
+ * \return The GWN.
+ */
+template <typename T>
+double winding_number(const Point<T, 2>& q,
+                      const axom::Array<detail::NURBSCurveGWNCache<T>>& c_arr,
+                      bool& isOnCurve,
+                      double edge_tol = 1e-8,
+                      double EPS = 1e-8)
+{
+  double gwn = 0;
+  isOnCurve = false;
+  for(int i = 0; i < c_arr.size(); ++i)
+  {
+    bool isOnThisCurve = false;
+    gwn += winding_number(q, c_arr[i], isOnThisCurve, edge_tol, EPS);
+    isOnCurve = isOnCurve || isOnThisCurve;
+  }
+
+  return gwn;
+}
+
+//! \brief Overload without optional return parameter
+template <typename T>
+double winding_number(const Point<T, 2>& q,
+                      const axom::Array<detail::NURBSCurveGWNCache<T>>& c_arr,
+                      double edge_tol = 1e-8,
+                      double EPS = 1e-8)
+{
+  bool dummy_isOnThisCurve = false;
+  return winding_number(q, c_arr, dummy_isOnThisCurve, edge_tol, EPS);
+}
+
+/*!
+ * \brief Computes the GWN for an array of 2D points wrt an array of cached data for 2D NURBS curves
+ *
+ * \param [in] q_arr The array of query point to test
+ * \param [in] c_arr The array of curve objects
+ * \param [in] edge_tol The physical distance level at which objects are considered indistinguishable
+ * \param [in] EPS Miscellaneous numerical tolerance level for nonphysical distances
+ * 
+ * \return The array of GWN values.
+ */
+template <typename T>
+axom::Array<double> winding_number(const axom::Array<Point<T, 2>>& q_arr,
+                                   const axom::Array<detail::NURBSCurveGWNCache<T>>& c_arr,
```

Comment:
I'm sure we can change this later, if necessary, but I wonder if the input arrays should be `axom::ArrayView` instead of `axom::ArrayView`.  That would give users more flexibility to call this w/ their own array types (other than `axom::Array`) without an explicit copy.